pkg/trace/sampler/exception_sampler.go

```diff
@@ -0,0 +1,207 @@
+package sampler
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/traceutil"
+	"golang.org/x/time/rate"
+)
+
+const (
+	// cardinalityLimit limits the number of spans considered per combination of (env, service).
+	cardinalityLimit = 1000
```

Comment:
How did we pick that value? What's the estimated extra memory usage? Should it be more/less?

Reply:
Figured it out looking at the shrink method.